components/motor/i2cmotors/ezopmp.go

```diff
@@ -292,8 +292,27 @@ func (m *Ezopmp) GoTo(ctx context.Context, mLPerMin, mins float64, extra map[str
 }
 
 // SetRPM instructs the motor to move at the specified RPM indefinitely.
-func (m *Ezopmp) SetRPM(ctx context.Context, rpm float64, extra map[string]interface{}) error {
-	return motor.NewSetRPMUnsupportedError(m.Name().ShortName())
+func (m *Ezopmp) SetRPM(ctx context.Context, mLPerMin float64, extra map[string]interface{}) error {
+	switch speed := math.Abs(mLPerMin); {
+	case speed < 0.1:
+		m.logger.CWarn(ctx, "motor speed is nearly 0 rev_per_min")
+		return motor.NewZeroRPMError()
+	case m.maxFlowRate > 0 && speed > m.maxFlowRate-0.1:
+		m.logger.CWarnf(ctx, "motor speed is nearly the max rev_per_min (%f)", m.maxFlowRate)
+	default:
+	}
+
+	ctx, done := m.opMgr.New(ctx)
+	defer done()
+
+	mins := math.Inf(1)
+	if math.Signbit(mLPerMin) {
+		mins = math.Inf(-1)
+	}
+
+	commandString := "DC," + strconv.FormatFloat(mLPerMin, 'f', -1, 64) + "," + strconv.FormatFloat(mins, 'f', -1, 64)
+	command := []byte(commandString)
```

Comment:
this might not actually work on the hardware, try this command from the [datasheet ](https://files.atlas-scientific.com/EZO_PMP_Datasheet.pdf)
![Screenshot 2024-06-18 at 16 11 27](https://github.com/viamrobotics/rdk/assets/35934754/6a3a603f-cd52-41b3-9b27-a281530299eb)

Reply:
will do, I'll test this PR next

Reply:
Actually, since we know the max flow rate of this motor, we can use that for an rpm -> power conversion and just call set power from SetRPM.